dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -1553,8 +1556,20 @@ protected synchronized void unsetVersion(ServiceReference<ServletVersion> refere
     
     public static int getServletContainerSpecLevel() {
         if (WebContainer.loadedContainerSpecLevel == SPEC_LEVEL_UNLOADED) {
-            logger.logp(Level.WARNING, CLASS_NAME, "getServletContainerSpecLevel", "servlet.feature.not.loaded.correctly");
-            return WebContainer.DEFAULT_SPEC_LEVEL;
+            CountDownLatch currentLatch = selfInit;
+            // wait for activation
+            try {
+                currentLatch.await(5, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                // auto-FFDC
+                Thread.currentThread().interrupt();
```

Comment:
Are you meaning to re-interrupt the thread or the remove the interrupt flag from the thread?  If the latter you will want to used the interrupted method instead of the interrupt method.

Reply:
Because I just catch the exception and do nothing I did intend to re-interrupt the flag here.  I am not sure what reason one would have to interrupt our thread, but I think it is best practice to keep the thread interrupted for cases like this.